### PR TITLE
Improve man pages + flags documentation

### PR DESCRIPTION
**Pull request by prashantv**
Add rules to generate the man page + HTML version, which we will push to a `gh-pages` branch so that the Github page is updated.

Also updated some benchmarking flags to use hyphens instead of camelCase which is the standard for flags.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,6 +26,14 @@ install_ci: install
 		go get golang.org/x/tools/cmd/cover
 
 
+.PHONY: update_man
+update_man:
+	go install .
+	yab --man-page > man/yab.1
```
Comment (blampe) [inline Makefile]: Do you need to `mkdir man`?

File: Makefile [hunk 2]
```diff
@@ -26,6 +26,14 @@ install_ci: install
...
+	nroff man/yab.1 -man -Thtml > man/yab.html
+	[[ -d ../yab_ghpages ]] && cp man/yab.html ../yab_ghpages/man.html
```
Comment (blampe) [inline Makefile]: Nice!

File: doc.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+by passing --health.
+
+Thrift requests can be specified as JSON or YAML. For example, for a method
+defined as:
+  void addUser(1: string name, 2: i32 age)
+
+You can pass the request as JSON: {"name": "Prashant", age: 10}
+or as YAML:
+  name: Prashant
+	age: 10
```
Comment (abhinav) [inline doc.go]: Indentation mismatch

File: Makefile (hunk 1 above)
Comment (prashantv) [inline Makefile]: This PR adds the `man` dir with the generated man pages so anyone running this command after this change will already have the directory

File: doc.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+The request can be specified on the command line using -r or --request:
+  $ yab -p localhost:9787 kv KeyValue::Get -3 '{"key": "hello"}'
```
Comment (abhinav) [inline doc.go]: You say `-r` and `--request` but you're using `-3`

File: doc.go [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+Or it can be loaded from a file using -f or --file:
+  $ yab -p localhost:9787 kv KeyValue::Get -f req.yaml
```
Comment (abhinav) [inline doc.go]: Also, `-m` was mentioned above and not used

File: doc.go [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+// It can be used in a curl-like fashion when benchmarking features are disabled.
+//
+// For usage information, check out the man page:
+// http://yarpc.github.io/yab/man.html
+package main
+
+const _reqOptsDesc = `Configures the request data and the encoding.
+
+To make Thrift requests, specify a Thrift file and pass the Thrift
+service and procedure to the method argument (-m) as Service::Method.
```
Comment (abhinav) [inline doc.go]: Prefer including the long form (`--method`) as well

File: doc.go (hunk 3 above)
Comment (prashantv) [inline doc.go]: 👍

File: doc.go (hunk 4 above)
Comment (prashantv) [inline doc.go]: wow, old habits die hard. Even when writing documentation I keep using `-3`. Changing to `-r`

File: doc.go [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+or
+  $ yab -p tchannel://localhost:9787 [options]
+
+For HTTP endpoints, specify the URL as the peer,
+  $ yab -p http://localhost:8080/thrift [options]
+
+The Thrift encoded body will be POSTed to the specified URL.
+
+Multiple peers can be specified using a peer list using -P or --peer-list.
+  $ yab -P hosts.json [options]
```
Comment (abhinav) [inline doc.go]: What's the behavior when you specify multiple peers?

File: doc.go [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+By default, yab will only make a single request. To enable benchmarking, you
+must specify the maximum duration for the benchmark by passing -d or --max-duration.
+
+yab will make requests till either the maximum requests (-n or --max-requests)
+or the maximum duration is reached.
+
+You can control the rate at which yab makes requests using the --rps flag.
+
+An example benchmark command might be:
+  yab -p localhost:9787 moe --health -n 100000 -d 10s --rps 1000
```
Comment (abhinav) [inline doc.go]: I would use the long forms of all options in this example (and some of the other ones). It explains exactly what each parameter means.

File: Makefile (hunk 1 above)
Comment (blampe) [inline Makefile]: Ah of course, overlooked that.

File: doc.go (hunk 6 above)
Comment (prashantv) [inline doc.go]: 👍

File: doc.go (hunk 7 above)
Comment (prashantv) [inline doc.go]: Good point, will add section on the behaviour.

File: doc.go (hunk 8 above)
Comment (prashantv) [inline doc.go]: The line becomes really long which actually ends up obscuring the useful info. I think `-d` and `-n` are pretty readable vs adding a lot of noise in `--max-duration` and `--max-requests`. I'd expect almost everyone to use the short form.

File: doc.go (hunk 5 above)
Comment (prashantv) [inline doc.go]: Added example with `-m` and explained how positional args can be used too. Trying to reduce the amount of noise.

File: doc.go (hunk 8 above)
Comment (abhinav) [inline doc.go]: 👍

File: doc.go [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+Thrift requests can be specified as JSON or YAML. For example, for a method
+defined as:
+  void addUser(1: string name, 2: i32 age)
+
+You can pass the request as JSON: {"name": "Prashant", age: 100}
+or as YAML:
+  name: Prashant
+  age: 100
+
+The request can be specified on the command line using -r or --request:
```
Comment (abhinav) [inline doc.go]: nit:

> The request body can [..]

**Comment by abhinav** [comment]
LGTM

File: doc.go (hunk 9 above)
Comment (prashantv) [inline doc.go]: fixed

**Pull request merged**